lib/src/grid_movies_events.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'widgets/homeScreenWidgets/grid_movies.dart';
+import 'state_type.dart';
+import 'ui/movies_details_page.dart';
+import 'utils/ui_constants.dart';
+
+class GridMoviesEvents extends StatelessWidget {
+  final StateType state;
+
+  const GridMoviesEvents({
+    required this.state,
+  });
+
+  Widget _buildContent(BuildContext context) {
+    switch (state.stateType) {
```

Comment:
Could we generate a new widget for each type of event that was launched (loading, success, etc), or see if they share the same structure?